lightning/src/ln/peer_handler.rs

```diff
@@ -1115,19 +1115,22 @@ impl<Descriptor: SocketDescriptor, CM: Deref> PeerManager<Descriptor, CM> where
 					return false;
 				}
 
+				let mut needs_to_write_data = false;
 				if let PeerState::Connected(ref mut conduit) = peer.encryptor {
 					let ping = msgs::Ping {
 						ponglen: 0,
 						byteslen: 64,
 					};
 					peer.pending_outbound_buffer.push_back(conduit.encrypt(&encode_msg!(&ping)));
+					needs_to_write_data = true;
+				}
 
+				if needs_to_write_data {
 					let mut descriptor_clone = descriptor.clone();
 					self.do_attempt_write_data(&mut descriptor_clone, peer);
-
-					peer.awaiting_pong = true;
 				}
 
+				peer.awaiting_pong = true;
```

Comment:
This seems to obfuscate the code. We don't need to await a pong until a ping is sent, so this could simply be:

```rust
if let PeerState::Connected(ref mut conduit) = peer.encryptor {
	let ping = msgs::Ping {
		ponglen: 0,
		byteslen: 64,
	};
	peer.pending_outbound_buffer.push_back(conduit.encrypt(&encode_msg!(&ping)));
	peer.awaiting_pong = true
	self.do_attempt_write_data(&mut descriptor.clone(), peer);
}
```

Arguably, the last few lines at some point may be refactored to simply:

```rust
peer.send_message(ping);
```

Reply:
alas, Rust 1.22.0 does not allow such simplicity. That's what it was originally.